File: app/src/main/java/ericdiaz/program/currencyconveterlive2019/repository/BaseRepository.java
package ericdiaz.program.currencyconveterlive2019.repository;

import androidx.annotation.NonNull;

import ericdiaz.program.currencyconveterlive2019.model.ExchangeRateResponse;
import io.reactivex.Single;

public interface BaseRepository {
    Single<ExchangeRateResponse> requestExchangeRates(@NonNull final String date,
                                                      @NonNull final String baseCurrency);
}

File: core/src/main/java/ericdiaz/program/data/repository/BaseRepository.kt
package ericdiaz.program.data.repository

import ericdiaz.program.data.model.CurrencyProfile
import ericdiaz.program.data.model.ExchangeRateResponse
import io.reactivex.Single

interface BaseRepository {
    fun requestExchangeRates(date: String, baseCurrency: String): Single<ExchangeRateResponse>

    fun requestCurrencyProfiles(): Single<Map<String, CurrencyProfile>>
}